### PR TITLE
enh: Use multilingual embedding model

### PR DESCRIPTION
**Pull request by marcelklehr**
*breaking change*

We're opting not to have a migration for this, instead people should recrawl all files.

We will need to clear the database though.

### CODE REVIEW

File: config.cpu.yaml [hunk 1]
```diff
@@ -34,11 +39,11 @@ embedding:
...
 
-  hugging_face:
+  #hugging_face:
     # model_name: all-MiniLM-L6-v2
-    model_name: sentence-transformers/all-mpnet-base-v2
-    model_kwargs:
-      device: cpu
+    #model_name: sentence-transformers/all-mpnet-base-v2
+    #model_kwargs:
+    #  device: cpu
```
Comment (kyteinsky) [inline config.cpu.yaml:46]: let's leave it uncommented, same for the gpu config

**Comment by kyteinsky** [comment]
> We're opting not to have a migration for this, instead people should recrawl all files.

let's not release it now until we also have the db migration then do a repair. It can save on a lot of compute for people who have indexed tons of files. Wdys?

File: config.cpu.yaml (hunk 1 above)
Comment (marcelklehr) [reply config.cpu.yaml:46]: Can't because the new model is hugging_face as well

**Comment by marcelklehr** [comment]
> let's not release it now until we also have the db migration then do a repair.

Yep, agreed

File: config.cpu.yaml (hunk 1 above)
Comment (kyteinsky) [reply config.cpu.yaml:46]: ah sorry didn't see it was inside embedding section. We can get rid of it then since it is the same thing as the top one.

**Comment by marcelklehr** [comment]
Shall we do a stable3 branch and then merge this into main?

File: config.gpu.yaml [hunk 2]
```diff
@@ -24,22 +24,21 @@ vectordb:
...
+  hugging_face:
+    model_name: nextcloud-ai/multilingual-e5-large-instruct
+    model_kwargs:
+      device: cuda
+
   instructor:
     model_name: hkunlp/instructor-base
     model_kwargs:
-      device: cuda
+      device: cpu
```
Comment (kyteinsky) [inline config.gpu.yaml:35]: cuda here

**Comment by marcelklehr** [comment]
Does LlamaCppEmbeddings also add a task prompt to the text to be embedded? We need that for the new model to perform at its best.

**Comment by kyteinsky** [comment]
> Does LlamaCppEmbeddings also add a task prompt to the text to be embedded? We need that for the new model to perform at its best.

Don't think we have that option here.

File: config.cpu.yaml [hunk 3]
```diff
@@ -24,6 +24,11 @@ vectordb:
     # url: http://localhost:8080
 
 embedding:
+  hugging_face:
+    model_name: nextcloud-ai/multilingual-e5-large-instruct
+    model_kwargs:
+      device: cpu
+
```
Comment (kyteinsky) [inline config.cpu.yaml:31]: `llama`, same as the gpu config

File: config.gpu.yaml [hunk 4]
```diff
@@ -24,19 +24,18 @@ vectordb:
     # url: http://localhost:8080
 
 embedding:
+  llama:
```
Comment (kyteinsky) [inline config.gpu.yaml:27]: not sure in the case of embedding mode, but does `n_gpu_layers: -1` make a difference?

**Comment by marcelklehr** [comment]
```
pydantic_core._pydantic_core.ValidationError: 2 validation errors for LlamaCppEmbeddings
model_name
  Extra inputs are not permitted [type=extra_forbidden, input_value='nextcloud-ai/multilingual-e5-large-instruct', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/extra_forbidden
model_kwargs
  Extra inputs are not permitted [type=extra_forbidden, input_value={'device': 'cpu'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.9/v/extra_forbidden
```

**Comment by kyteinsky** [comment]
`model_name` and `model_kwargs` are options for huggingface in config.cpu.yaml.
The gpu config has the correct config.
```yaml
embedding:
  llama:
    model_path: multilingual-e5-large-instruct-q6_k.gguf
    n_batch: 512
    n_ctx: 8192
```

**Comment by marcelklehr** [comment]
:tada: It's working

**Pull request merged**